### PR TITLE
fix(table): optimize type declaration about filter

### LINKED ISSUE #28971: Mismatch between the type of TableProps::onChange::filters and ColumnType:filteredValue
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Steps to reproduce
The TableProps has an optional function `onChange` and the param filters can be `Record<string, (Key | boolean)[] | null>`, where `Key` is `string | number`.
```
export interface TableProps<RecordType> extends Omit<RcTableProps<RecordType>, 'transformColumns' | 'internalHooks' | 'internalRefs' | 'data' | 'columns' | 'scroll' | 'emptyText'> {
   [...]
    onChange?: (pagination: TablePaginationConfig, filters: Record<string, (Key | boolean)[] | null>
```

However, the field `filteredValue` in the interface `ColumnType` can only be `Key`, `string | number`. 
```
export interface ColumnType<RecordType> extends RcColumnType<RecordType> {
    filteredValue?: Key[] | null;
```

### What is expected?
In my old code, I used to get the filters from `onChange` and just assign them to `filteredValue`. I don't understand how can they have different types if they are tied together, one is generated by the other. I expected this to still work after the update.

### What is actually happening?
They have different types so you can't assign the result of one as a parameter to the other, because `filters` not also allows `boolean`, but `filteredValue` doesn't. 

| Environment | Info |
|---|---|
| antd | 4.10.3 |
| React | 16.13.1 |
| System | Windows 10 |
| Browser | Chrome 88.0.4324.96 |

---
I didn't add a codepen URL, I wasn't able to properly use antd there (I keep getting the error that antd is not defined). Let me know is this is really needed.

### PR DESCRIPTION
**Pull request by mumiao**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [x] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link
closed https://github.com/ant-design/ant-design/issues/28971

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Optimize Table type declaration about filter.  |
| 🇨🇳 Chinese | 优化表格 filter 相关类型声明。         |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by mumiao** [comment]
@kerm1it

File: components/table/interface.tsx [hunk 1]
```diff
@@ -68,6 +68,8 @@ export type ColumnTitle<RecordType> =
   | React.ReactNode
   | ((props: ColumnTitleProps<RecordType>) => React.ReactNode);
 
+export type FilterValues = (Key | boolean)[];
```
Comment (kerm1it) [inline components/table/interface.tsx:71]:
```suggestion
export type FilterValue = (Key | boolean)[];
```

File: components/table/interface.tsx [hunk 2]
```diff
@@ -68,6 +68,8 @@ export type ColumnTitle<RecordType> =
...
+export type FilterKeys = Key[] | null;
```
Comment (kerm1it) [inline components/table/interface.tsx:72]:
```suggestion
export type FilterKey = Key[] | null;
```

File: components/auto-complete/index.tsx [hunk 3]
```diff
@@ -54,7 +54,6 @@ const AutoComplete: React.ForwardRefRenderFunction<RefSelectProps, AutoCompleteP
   }
 
   const getInputElement = customizeInput ? (): React.ReactElement => customizeInput! : undefined;
-
```
Comment (kerm1it) [inline components/auto-complete/index.tsx:57]: 这里为啥要删除这一行？

Comment (mumiao) [reply components/auto-complete/index.tsx:57]: lint-fix 自动跑的

Comment (kerm1it) [reply components/auto-complete/index.tsx:57]: 不应该呀，我本地并么有报错，其他的地方还有类似的写法都没有问题呀。

Comment (mumiao) [reply components/auto-complete/index.tsx]: 可能是我误删了 分界线我恢复了  应该没问题了

**Review by kerm1it — APPROVED** [review]

**Pull request merged**